3p/ampcontext.js

```diff
@@ -24,7 +24,10 @@ export class AmpContext {
    *  @param {Window} win The window that the instance is built inside.
    */
   constructor(win) {
-    this.setupMetadata_();
+    this.win_ = win;
+    if (!this.getWindowSentinel_()) {
+      this.setupMetadata_();
```

Comment:
can you add some doc to explain in what scenario we don't need to parse the metadata?